[REACH-344] feat: Start on the Landing Page Section area in Gamut

judahtanthony opened the pull request:
## Overview

### PR Checklist

- [x] Related to designs: https://www.figma.com/file/2aBlW1pgFt5PT6soRu1HXy/%E2%9C%8F%EF%B8%8F-Landing-Page-Templates---Unification?node-id=1%3A43
- [x] Related to JIRA ticket: REACH-344
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### Description
These are just some of the smaller entities that be used in a number of Landing Page Section organisms.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/client-modules?pullRequest=1118) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/client-modules?pullRequest=1118) before we can accept your contribution.<br/><hr/>**Judah Anthony** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/client-modules?pullRequest=1118) it.</sub>

JoshuaKGoldberg commented on packages/gamut-labs/src/brand/LPSection/Title.tsx:
```diff
@@ -0,0 +1,37 @@
+
+import { Heading } from '@codecademy/gamut';
+
+// Heading seems to not take className.
```
Err, it does: `HeadingProps` takes it from `ContainerElementProps`. What odd behavior are you seeing?

judahtanthony replied:
Hmmm.  I don't know why I thought that.  It is not true.  I will remove this comment.

judahtanthony replied:
removed.

JoshuaKGoldberg commented on packages/gamut-labs/src/brand/LPSection/Title.tsx:
```diff
@@ -0,0 +1,37 @@
+  testId?: string;
+};
+
+export const LPSectionTitle: React.FC<LPSectionTitleProps> = ({
```
I'm averse to including abbreviations that aren't super duper common across the industry, as they can make the code harder to read for people not already immersed in our particular structures. Can section be `LandingPage*` instead of `LP*`, please?

judahtanthony replied:
Will do.

judahtanthony replied:
updated.

JoshuaKGoldberg commented on packages/gamut-labs/src/brand/LPSection/Title.tsx:
```diff
@@ -0,0 +1,37 @@
+    } : {
+      lg: 'lg',
+      xs: 'md',
+    }}
```
Oh man @codecaaron this is, like, a perfect example of a variant like your system is trying to solve, at the props level 😄

JoshuaKGoldberg commented on packages/gamut-labs/src/brand/LPSection/CTA.tsx:
```diff
@@ -0,0 +1,19 @@
+  href,
+  children,
+}) => (
+  <CTA theme="hyper" href={href}>{children}</CTA>
```
Yeah the `theme` prop is a tricky one: it's somewhat reserved by Emotion. Hence this very bizarre [CI error](https://app.circleci.com/pipelines/github/Codecademy/client-modules/7636/workflows/c6fb3e3f-c021-47b4-a02c-27d3f2316542/jobs/44438):

```
$ tsc --emitDeclarationOnly
src/brand/LPSection/CTA.tsx(18,8): error TS2322: Type 'string' is not assignable to type 'undefined'.
```

You'll want to use some other name instead. `variant` is what I've been suggesting to folks.

judahtanthony replied:
Oooohhh.  Thanks.  Should we update [Button](https://gamut.codecademy.com/storybook/?path=/docs/atoms-button--button) (which uses theme) to alias that prop?  I'm currently just letting Emotion to proxy all the props through.

JoshuaKGoldberg replied:
Yeah I was thinking about doing that... we're about to deprecate it (https://github.com/codecademy-engineering/Codecademy/pull/20004 -> https://github.com/Codecademy/client-modules/pull/1069) so after that goes in I can send a `variant` prop alias.

judahtanthony replied:
For now I just wrapped the raw button.  So far we just need simple layout-y things.

JoshuaKGoldberg reviewed (changes requested):
Glorious start! Nice to see usage of the new styled components and mdx stuff. I can take a deeper look once it builds 👍

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.6.3-alpha.462e37.0<br/>@codecademy/styleguide@17.10.4-alpha.462e37.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fa438475d566e0a86eeaa91--gamut-preview.netlify.app>https://5fa438475d566e0a86eeaa91--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fa438475d566e0a86eeaa91>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

JoshuaKGoldberg commented on packages/gamut-labs/src/brand/LandingPageSection/CTA.tsx:
```diff
@@ -0,0 +1,25 @@
+
+import { Button } from '@codecademy/gamut';
+
+const CTA = styled('div')`
```
⛳ ✨ 

```suggestion
const CTA = styled.div`
```

rogervera replied:
@JoshuaKGoldberg Birdie!

judahtanthony replied:
Updated.

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.6.3-alpha.ad8961.0<br/>@codecademy/styleguide@17.10.4-alpha.ad8961.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fa43a5c23b839009f8de849--gamut-preview.netlify.app>https://5fa43a5c23b839009f8de849--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fa43a5c23b839009f8de849>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

rogervera commented on packages/gamut-labs/src/brand/LandingPageSection/Description.tsx:
```diff
@@ -0,0 +1,24 @@
+
+import { Markdown } from '@codecademy/gamut';
+
+const Description = styled('div')`
```
Suggestion for `styled.div` same as https://github.com/Codecademy/client-modules/pull/1118/files#r518240948

judahtanthony replied:
Thanks.  Fixed.

rogervera reviewed (approved):
LGTM!
Per your post in Slack, I do agree that I see these more as atoms than organisms however.

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.6.3-alpha.3789e0.0<br/>@codecademy/styleguide@17.10.4-alpha.3789e0.0<br/></pre><!--LERNA_VERSION_COMMENT-->

judahtanthony commented:
> Glorious start! Nice to see usage of the new styled components and mdx stuff. I can take a deeper look once it builds 👍

I think I have addressed all your concerns.

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fa4452e662f63111b4c35c6--gamut-preview.netlify.app>https://5fa4452e662f63111b4c35c6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fa4452e662f63111b4c35c6>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

judahtanthony commented:
> LGTM!
> Per your post in Slack, I do agree that I see these more as atoms than organisms however.

100%
I ultimately would just like a LandingPageSection "gallery" for our marketers, so the shouldn't even see thing like "title" which they cannot directly choose.

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.6.3-alpha.bb1432.0<br/>@codecademy/styleguide@17.10.4-alpha.bb1432.0<br/></pre><!--LERNA_VERSION_COMMENT-->

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fa44855ccaccc104fc90c66--gamut-preview.netlify.app>https://5fa44855ccaccc104fc90c66--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fa44855ccaccc104fc90c66>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.8.0-alpha.21fc6d.0<br/>@codecademy/styleguide@17.12.0-alpha.21fc6d.0<br/></pre><!--LERNA_VERSION_COMMENT-->

JoshuaKGoldberg commented on packages/gamut-labs/src/brand/LandingPageSection/CTA.tsx:
```diff
@@ -0,0 +1,27 @@
+  children,
+}) => (
+  <CTA className={className} data-testid={testId}>
+    <Button theme="hyper" href={href}>
```
Just confirming: this is a _start_, right? Because this button is kind of useless without an `onClick` 😄 .

JoshuaKGoldberg commented on packages/gamut-labs/src/brand/LandingPageSection/CTA.tsx:
```diff
@@ -0,0 +1,27 @@
+
+import { Button } from '@codecademy/gamut';
+
+const CTA = styled.div`
```
Could you put the margin on the button? `styled(Button)`?

JoshuaKGoldberg commented on packages/styleguide/stories/Labs/Brand/Organisms/LandingPageSectionTitle.stories.mdx:
```diff
@@ -0,0 +1,27 @@
+import { LandingPageSectionTitle } from '@codecademy/gamut-labs/src';
+
+<Meta
+  title="Labs/Brand/Organisms/LandingPageSectionTitle"
```
I agree, these should really be atoms. You could always add an organism section afterwards for things that contain them, but these aren't it.

JoshuaKGoldberg reviewed (approved):
LGTM, with the caveats requested :)

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fa44a12d9e83810d0af06e0--gamut-preview.netlify.app>https://5fa44a12d9e83810d0af06e0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fa44a12d9e83810d0af06e0>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->
